src/middleware/node/middleware.ts

```diff
@@ -1,8 +1,19 @@
 // remove type imports from http for Deno compatibility
 // see https://github.com/octokit/octokit.js/issues/24#issuecomment-817361886
 // import { IncomingMessage, ServerResponse } from "http";
-type IncomingMessage = any;
```

Comment:
Despite this `any`, we did clearly have some implicit assumptions before about what properties the `IncomingMessage` would have. Is the interface you define below common across many JavaScript libraries/frameworks?

Reply:
They aren't really "implicit assumptions", the middleware was designed for NodeJS HTTP servers of the likes of express, and raw `http.Server` implementations.

We had to comment out the imports due to Deno compatibility (The link in the comment is out of date)
See https://github.com/octokit/octokit.js/issues/2075#issuecomment-817361886 for context.
That's why they are currently set to `any`